manifests/plugin/discovery.pp

```diff
@@ -17,22 +17,18 @@
 #                    type:String
 #
 class foreman_proxy::plugin::discovery (
-  $install_images = $::foreman_proxy::plugin::discovery::params::install_images,
-  $tftp_root      = $::foreman_proxy::plugin::discovery::params::tftp_root,
-  $source_url     = $::foreman_proxy::plugin::discovery::params::source_url,
-  $image_name     = $::foreman_proxy::plugin::discovery::params::image_name,
+  Boolean $install_images         = $::foreman_proxy::plugin::discovery::params::install_images,
+  Stdlib::Absolutepath $tftp_root = $::foreman_proxy::plugin::discovery::params::tftp_root,
```

Comment:
I had the same question in my mind. But the params.pp provides working default settings so in my opinion the datatype is okay here.